### DISCUSSION TITLE
Show Items by CategoryId on Dynamic Page (nextjs, prisma, typescript)

### DISCUSSION THREAD

**Discussion by peterparker2005** [prisma/prisma#19486] [Q&A]
Hello, I think I have almost everything, but I don't really know how to show items by categoryId on dynamic page. I have these parts of code: 

**Prisma model >>**
```prisma
model Item {
  id         Int      @id @default(autoincrement())
  slug       String
  title      String
  about      String
  category   Category @relation(fields: [categoryId], references: [id])
  categoryId Int
  img_path   String
}

model Category {
  id       Int    @id @default(autoincrement())
  slug     String
  title    String
  items    Item[]
  img_path String
}
```
**Func to get items by category (thanks Nurul) >>**
```js
export async function getCategoryById(id: number) {
  try {
    const category = await prisma.category.findUnique({
      where: { id },
      include: {
        items: true,
      },
    });
    return { category };
  } catch (error) {
    return { error };
  }
}
```

**Block where I link to the [category] page >>**
```js
import Link from 'next/link';
import CategoryBlock from './CategoryBlock';
import { getCategories, getCategoryById } from '@/prisma/categories';

const CategoryPage = async () => {
  const { categoriesFound } = await getCategories();

  return (
    <main className="flex flex-col gap-10 w-full">
      <p className="font-medium text-3xl">Категории</p>
      <div className="grid grid-cols-4 gap-x-5 gap-y-10 w-full">
        {categoriesFound?.map((category) => {
          return (
            <Link href={`/catalog/${category.id}`} key={category.id}>
              <CategoryBlock
                src={`/category/${category.img_path}.png`}
                alt={category.title}
                title={category.title}
              />
            </Link>
          );
        })}
      </div>
    </main>
  );
};

export default CategoryPage;

```

**[category] Page itself (need to do something here) >>**
```js
const ItemsListByCategory = () => {
  return (
    <div className="min-h-screen mt-40 container">ItemsListByCategory</div>
  );
};
export default ItemsListByCategory;
```

Important note that Items should be on _server-side_ I think (maybe I'm wrong)...

**Comment by ludralph** [comment]
Hi @coutp 👋 

To show items by `categoryId` on the dynamic page, you can modify your `[category]` page to fetch the items of the selected category using the `getCategoryById` function. Here's an example:

```js
export async function getServerSideProps(context) {
  const { category } = await getCategoryById(Number(context.query.category));
  
  if (!category) {
    return {
      notFound: true,
    };
  }

  return {
    props: { category },
  };
}
```
we use `getServerSideProps` to fetch the category and its associated items based on the `category` parameter in the route. The `category` object is then passed as a prop to the ItemsListByCategory component, which renders the items on the page.


If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by ludralph** [reply]
Hi @coutp 

do you have any more questions around this issue? If no, it would be really helpful if you could quickly **mark this Discussion as _answered_** to indicate that your issue has been resolved. 

In case you do have any outstanding questions, feel free to let us know and we'd be happy to help!

**Reply by nurul3101** [reply]
@coutp I am checking to see if you had a chance to look at Raphael's last response. Please let us know in case you still need assistance.

**Reply by nurul3101** [reply]
We are closing this Discussion due to inactivity. Please open a new Discussion if you have a similar problem 🙏
